src/components/ec-smart-table/ec-smart-table.vue

```diff
@@ -260,4 +280,38 @@ function getEcTableSlots() {
   delete tableSlots.pages;
   return tableSlots;
 }
+
+// infinite scroll
+
+const intersectionTarget = ref(null);
+
+const canLoadMore = computed(() => props.data && props.data.length < props.totalRecords);
```

Comment:
are we going to pass all data to make this computed false?

For example we have 100 total records, and 10 per page, to stop intersection we need to pass all 100 records to `props.data`, and in this case we'll display all 100 records in the table.

What if user have 1000, 5000 totalRecords, we'll paste them all in our table and DOM?

Reply:
Sadly yes, I know that in terms of performance is going to be awful when we have thousand of records and the user wants to scroll all the way down. 

Even if we want to remove from DOM the ones that were already scrolled (let's say that in the screen you can only fit 10 records, in theory we can remove the ones outside the viewport from DOM) we still need to store them in memory so I don't think this would be helpful at all.

It's possible to implement a kind of 2-way infinite scrolling (when scrolling down but also when scrolling up) to avoid that, but:
- I think is over-engineering for our purpose, at least for now
- In terms of UX is even worse than the normal infinite scrolling

Reply:
Even though I can accept data storing for all fetched records, which is also debatable. I think we should prevent to render lots of records setting maximum of rendered rows. On some env we have thousands of records for this table, as I understand it's normal situation.